File: backend/models/Menu.js
const mongoose = require('mongoose');

const menuSchema = new mongoose.Schema({
  nombre: { type: String, required: true },
  descripcion: String,
  precio: { type: String, required: true },  // El precio está como String en la base de datos
  restaurante_id: { type: mongoose.Schema.Types.ObjectId, ref: 'Restaurantes', required: true }  // Referencia al Restaurante
}, { timestamps: true });

module.exports = mongoose.model('Menu', menuSchema, 'Menu');  // 'Menu' es el nombre de la colección en la base de datos

File: backend/app/models/menu.py
# models/menu.py

from pydantic import BaseModel, Field
from typing import Optional

class PlatilloIn(BaseModel):
    nombre: str
    descripcion: str
    precio: float
    restaurante_id: str

class PlatilloOut(PlatilloIn):
    id: str = Field(..., alias="_id")

class PlatilloUpdate(BaseModel):
    nombre: Optional[str] = None
    descripcion: Optional[str] = None
    precio: Optional[float] = None
    restaurante_id: Optional[str] = None
